src/meta/src/hummock/manager/mod.rs

```diff
@@ -878,259 +876,319 @@ impl HummockManager {
-                for table_id in &compact_task.existing_table_ids {
-                    table_to_vnode_partition.insert(*table_id, vnode_partition_count);
+                let is_trivial_reclaim = CompactStatus::is_trivial_reclaim(&compact_task);
+                let is_trivial_move = CompactStatus::is_trivial_move_task(&compact_task);
+                if is_trivial_reclaim || (is_trivial_move && can_trivial_move) {
+                    let log_label = if is_trivial_reclaim {
+                        "TrivialReclaim"
+                    } else {
+                        "TrivialMove"
+                    };
+                    let label = if is_trivial_reclaim {
+                        "trivial-space-reclaim"
+                    } else {
+                        "trivial-move"
+                    };
+
+                    tracing::debug!(
+                        "{} for compaction group {}: input: {:?}, cost time: {:?}",
+                        log_label,
+                        compact_task.compaction_group_id,
+                        compact_task.input_ssts,
+                        start_time.elapsed()
+                    );
+                    compact_task.set_task_status(TaskStatus::Success);
+                    compact_status.report_compact_task(&compact_task);
+                    if !is_trivial_reclaim {
+                        compact_task.sorted_output_ssts =
+                            compact_task.input_ssts[0].table_infos.clone();
+                    }
+                    self.metrics
+                        .compact_frequency
+                        .with_label_values(&[
+                            label,
+                            &compact_task.compaction_group_id.to_string(),
+                            selector.task_type().as_str_name(),
+                            "SUCCESS",
+                        ])
+                        .inc();
+                    let version_delta = gen_version_delta(
+                        &mut hummock_version_deltas,
+                        &mut branched_ssts,
+                        &current_version,
+                        &compact_task,
+                        deterministic_mode,
+                    );
+                    current_version.apply_version_delta(&version_delta);
+                    trivial_tasks.push(compact_task);
+                    trivial_move_count += 1;
+                    if trivial_move_count >= MAX_TRIVIAL_MOVE_TASK_COUNT {
+                        break;
+                    }
+                } else {
+                    let mut table_to_vnode_partition = match self
+                        .group_to_table_vnode_partition
+                        .read()
+                        .get(&compaction_group_id)
+                    {
+                        Some(table_to_vnode_partition) => table_to_vnode_partition.clone(),
+                        None => BTreeMap::default(),
+                    };
+                    table_to_vnode_partition
```

Comment:
I think we can clear it every time before we call this method

Reply:
> I think we can clear it every time before we call this method

I understand the reason why we don't need to clear it is because we directly modify `compact_task.table_vnode_partition` in this PR.

But there are still issues here:
1. The local variable `table_to_vnode_partition` is unused in this PR. Do we forget to use it in some code path?
2. We won't set `compact_task.table_vnode_partition` if `split_weight_by_vnode == 0` while prior to this PR we will set it. Is this expected?

Reply:
It is a mistake, I have fix it